### DISCUSSION TITLE
Is there a way to re-create all events?

### DISCUSSION THREAD

**Discussion by leandronsp** [RailsEventStore/rails_event_store#1133] [General]
Hi,

Maybe I didn't understand quite well some aspects of event sourcing, but I do guess that I can re-create all my data by simply running my events again. 

Scenario:

* I have an event called `UserCreated`
* I have a subscriber called `UserReadModelSubscriber` which subscribes to the event `UserCreated`, then creates the `User` on another table
* Table `Events` has 5 rows
* Table Users has 5 rows 

I'd like to remove all Users (`User.destroy_all`) then re-create them by running something like:
```
events = event_store.read.to_a

# for each event
event_store.publish(event)
```
Of course I get the error `EventDuplicatedInStream`, but I don't know if I have this capability on this gem.

Thanks for any help!

**Comment by jandudulski** [comment]
You don't want to re-publish the events - that would cause to handle them by every subscriber, even if they don't need to rebuild the state.

You should do something as simple as `event_store.read.each { |event| UserReadModelSubscriber.new.call(event) }`

**Comment by leandronsp** [comment]
Great! That's exactly what I needed, thanks!
